### PR TITLE
Customizable thresholds & accepts explicit options.units

### PR DESCRIPTION
**Pull request by rxaviers**
The algorithm had a big change (which is now more generic), but the default behavior is still the same.

### CODE REVIEW

File: test/relative-time.js [hunk 1]
```diff
@@ -16,80 +16,115 @@ describe("relative-time", function() {
...
+    });
+
+    it("should format years-distant dates", function() {
+      expect(relativeTime.format(new Date("2010-06-01 12:00"))).to.equal("6 years ago");
+      expect(relativeTime.format(new Date("2015-12-31 23:59"))).to.equal("last year");
+      expect(relativeTime.format(new Date("2017-01-01 00:00"))).to.equal("next year");
+
+      sinon.useFakeTimers(new Date("2016-10-02 12:00").getTime());
+      expect(relativeTime.format(new Date("2017-01-01 00:00"))).to.equal("next year");
+    });
```
Comment (rxaviers) [inline test/relative-time.js]: This is best reviewed by ignoring white-space change... Note the only actual change was the "best-fit" describe and this corrected test:

```diff
- expect(relativeTime.format(new Date("2016-04-10 11:00:01"))).to.equal("59 minutes ago");
+ expect(relativeTime.format(new Date("2016-04-10 11:01:00"))).to.equal("59 minutes ago");
```

File: src/relative-time.js [hunk 2]
```diff
@@ -1,94 +1,136 @@
...
- * [1d, 2d[ - yesterday
- * [xd, x+1d[ - x days ago
- * [xd, x+1d[ - last month if x >= 7 and range is on different months.
- * [1month, 2month[ - last month
- * [xmonth, x+1month[ - x months ago
- * [xmonth, x+1month[ - last year if x >= 3 and range is on different years.
- * [1y, 2y[ - last year
- * [xy, x+1y[ - x years ago
- *
- */
```
Comment (rxaviers) [inline src/relative-time.js]: Exporting this to README.md in a more clear way...

File: README.md [hunk 3]
```diff
@@ -25,6 +69,164 @@ console.log(relativeTime.format(new Date()));
...
+a: this year / 0 years ago
+b: last year / 1 year ago
+c: last year / 1 year ago
+d: 2 years ago
+e: 2 years ago
+f: 3 years ago
+g: 3 years ago
+```
+
+## Note that (although not shown by the above ruler), the years distances doesn't match weeks distance or days distance uniformly.
```
Comment (jzaefferer) [inline README.md]: Is this supposed to be a header? Looks pretty wrong to me.

Comment (rxaviers) [inline README.md]: This shouldn't be a header, thanks

**Comment by rxaviers** [comment]
> Reviewing the new algorithm in detail will take too much time that I don't have. Since you're reusing all the tests that I've reviewed before (and contributed to), I'm approving this change 👍

Thanks

**Comment by rxaviers** [comment]
Merged by 87d3255c58a9776bc806b22281208a5b72e5dd0b

**Pull request closed without merging**